### PR TITLE
feat(state): add `rxState` function

### PR DESCRIPTION
**Pull request by edbzn**
The follow-up to #1585, introduce functional `rxState` leveraging the inject function.

### CODE REVIEW

File: apps/docs/docs/state/setup.mdx [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+```typescript
+import { rxState } from '@rx-angular/state';
+import { RxFor } from '@rx-angular/template/for';
+
+@Component({
+  template: `<movie *rxFor="let movie of movies$" [movie]="movie" />`,
+  imports: [RxFor],
+})
+export class MovieListComponent {
+  private store = inject(Store<MovieState>);
```
Comment (eneajaho) [inline apps/docs/docs/state/setup.mdx:60]: Type inference doesn't work correctly with inject. 
```suggestion
  private store = inject<Store<MovieState>>(Store);
```

Comment (eneajaho) [reply apps/docs/docs/state/setup.mdx:60]: Or we remove the generic type completly

File: apps/docs/docs/state/setup.mdx [hunk 2]
```diff
@@ -0,0 +1,645 @@
...
+import { RxIf } from '@rx-angular/template/if';
+import { map, catchError, startWith, endWith } from 'rxjs';
+
+@Component({
+  template: `
+    <loader *rxIf="loading$" />
+    <error *rxIf="error$" />
+    <movie *rxFor="let movie of movies$" [movie]="movie" />
+  `,
+  imports: [RxFor, RxIf],
```
Comment (eneajaho) [inline apps/docs/docs/state/setup.mdx:92]: Shouldn't we add loader, error and movie components in the imports for the example to work correctly when copied?

File: apps/docs/docs/state/setup.mdx [hunk 3]
```diff
@@ -0,0 +1,645 @@
...
+    <div>{{ count$ | async }}</div>
+    <button (click)="increment.next()">increment</button>
+    <button (click)="decrement.next()">decrement</button>
+  `,
+})
+export class CounterComponent {
+  readonly count$ = this.state.select('count');
+
+  //highlight-next-line
+  @Input() set count(count: Observable<number> | number) {
```
Comment (eneajaho) [inline apps/docs/docs/state/setup.mdx:204]: Can we do this ? 
```suggestion
  @Input({ transform: coerceObservable }) set count(count: Observable<number> | number) {
```

File: apps/docs/docs/state/setup.mdx [hunk 4]
```diff
@@ -0,0 +1,645 @@
...
+    // decrement
+    connect('count', this.decrement, ({ count }) => count--);
+  });
+}
+```
+
+## Updates based on previous state
+
+Often it is needed calculate your new state based off some input and a previous state. The following example shows a
+local filtering alogrithm implemented with `RxState`.
```
Comment (eneajaho) [inline apps/docs/docs/state/setup.mdx:275]:
```suggestion
local filtering algorithm implemented with `RxState`.
```

File: apps/docs/docs/state/setup.mdx [hunk 5]
```diff
@@ -0,0 +1,645 @@
...
+
+Create a local `Service` by using `rxState` as factory function.
+
+```typescript
+import { InjectionToken, inject } from '@angular/core';
+import { rxState } from '@rx-angular/state';
+
+export const MovieState = new InjectionToken({
+  factory: () => {
+    // inject dependencies here
```
Comment (eneajaho) [inline apps/docs/docs/state/setup.mdx:404]: We can say also: Inject dependencies outside rxState

File: libs/state/src/lib/inject.ts
Comment (eneajaho) [inline libs/state/src/lib/inject.ts:1]: the name of the file is inject.ts? Shouldn't it be rx-state.ts ?

File: apps/docs/docs/state/setup.mdx (hunk 2 above)
Comment (edbzn) [reply apps/docs/docs/state/setup.mdx:92]: Then we should add all missing properties like standalone, selector, imports... I would suggest keeping the example minimal.

**Pull request merged**